src/main/java/seedu/address/model/person/StudentNumber.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's student number in the class group.
+ * Guarantees: immutable; is valid as declared in {@link #isValidStudentNumber(String)}
+ */
+public class StudentNumber {
+
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Student Number numbers should begin with a capital 'A', followed by 7 digits"
+                    + " and finally concluded by one capitalised letter.";
+    public static final String VALIDATION_REGEX = "^[A][0-9]{7}[A-Z]";
```

Comment:
Nice, why don't we allow the user to ignore case when entering the "/a" field? The regex match here enforces the user to enter the student number with a capitalized "A". Instead, we can do something like `^[Aa][0-9]{7}[A-Z]` to allow for more flexibility? 

Just a suggestion :)

Reply:
The last letter can be [a-zA-Z] to include lowercaps too